app/src/main/java/org/dhis2/usescases/enrollment/EnrollmentActivity.kt

```diff
@@ -340,8 +340,17 @@ class EnrollmentActivity : ActivityGlobalAbstract(), EnrollmentView {
             binding.teiDataHeader.mainAttributes.text =
                 String.format("%s %s", firstAttr, secondAttr)
             binding.teiDataHeader.secundaryAttribute.text = thirdAttr
+
+            if(firstAttr.isEmpty() && secondAttr.isEmpty()){
+                binding.teiDataHeader.mainAttributes.visibility = View.GONE
+            }
+            if(thirdAttr.isEmpty()){
+                binding.teiDataHeader.secundaryAttribute.visibility = View.GONE
+            }
```

Comment:
this is small but I usually would have put it in the presenter method that handles the view logic.

Reply:
That will make it a bigger change. So maybe we can do it next time

Reply:
I had approve it the PR. Won't do it next time...

The logic of the view should always go to the presenter as a norm, that is the presenter responsibility. You could have 2 methods in the activity and call them from the presenter. In addition, you could have your happy unit tests.